core/src/main/scala/kafka/api/FetchRequest.scala

```diff
@@ -196,12 +200,14 @@ case class FetchRequest(versionId: Short = FetchRequest.CurrentVersion,
   }
 
   override  def handleError(e: Throwable, requestChannel: RequestChannel, request: RequestChannel.Request): Unit = {
-    val fetchResponsePartitionData = requestInfo.map { case (topicAndPartition, _) =>
-      (topicAndPartition, FetchResponsePartitionData(Errors.forException(e).code, -1, MessageSet.Empty))
+    val responseData = new util.LinkedHashMap[TopicPartition, JFetchResponse.PartitionData]
```

Comment:
Even better than a test, it won't compile if someone changes it. :) I added a commit that removes the explicit constructors for version 0, 1 and 2 since they were only used in one place outside of tests and that place could be simplified by using the constructor that takes the version as a parameter.